### PR TITLE
Add default batch scheduler argument

### PR DESCRIPTION
**Pull request by jacobsalway**
## Purpose

Allow users to specify a default batch scheduler for all applications on the cluster. This allows all applications to benefit from batch scheduling features e.g. gang scheduling without needing to change the spec of all applications.

Completes the second task in https://github.com/kubeflow/spark-operator/issues/2098

## Change Category
Indicate the type of change by marking the applicable boxes:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that could affect existing functionality)
- [ ] Documentation update

## Checklist
Before submitting your PR, please review the following:

- [X] I have conducted a self-review of my own code.
- [ ] I have updated documentation accordingly.
- [ ] I have added tests that prove my changes are effective or that my feature works.
- [X] Existing unit tests pass locally with my changes.

### CODE REVIEW

**Comment by jacobsalway** [comment]
Thanks @ChenYi015 forgot about the Helm unit tests. Will make sure to add for any future Helm changes

**Review by ChenYi015 — APPROVED** [review]

**Comment by ChenYi015** [comment]
LGTM, will wait for another approval @vara-bonthu @yuchaoran2011.

**Review by vara-bonthu — APPROVED** [review]
Nice work on this @jacobsalway!  
/lgtm

**Pull request merged**
